pkg/slayers/path/scion/raw_spec.gobra

```diff
@@ -481,6 +481,60 @@ pure func CurrSeg(raw []byte,
 		segment(raw, offset, currHfIdx, ainfo, uinfo, consDir, peer, segLen)
 }
 
+ghost
+opaque
+requires 0 < segLen
+requires 0 <= currHfIdx && currHfIdx <= segLen
+requires segLen*path.HopLen == len(hopfields)
+requires acc(sl.AbsSlice_Bytes(hopfields, 0, len(hopfields)), R56)
+decreases
+pure func CurrSegWithInf(hopfields []byte, currHfIdx int, segLen int, inf path.IntermediateAbsInfoField) io.IO_seg3 {
+	return segment(hopfields, 0, currHfIdx, inf.AInfo, inf.UInfo, inf.ConsDir, inf.Peer, segLen)
+}
+
+ghost
+requires path.InfoFieldOffset(currInfIdx, MetaLen) + path.InfoLen <= offset
+requires 0 < segLen
+requires offset + path.HopLen * segLen <= len(raw)
+requires 0 <= currHfIdx && currHfIdx <= segLen
+requires 0 <= currInfIdx && currInfIdx < 3
+preserves acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), R50)
+preserves acc(sl.AbsSlice_Bytes(raw[offset:offset+segLen*path.HopLen], 0, segLen*path.HopLen), R50)
+preserves let infOffset := MetaLen + currInfIdx*path.InfoLen in
+	acc(sl.AbsSlice_Bytes(raw[infOffset:infOffset+path.InfoLen], 0, path.InfoLen), R50)
+ensures let infOffset := MetaLen + currInfIdx*path.InfoLen in
+	let inf := path.BytesToIntermediateAbsInfoField(raw[infOffset:infOffset+path.InfoLen], 0, 0, path.InfoLen) in
+	CurrSegWithInf(raw[offset:offset+segLen*path.HopLen], currHfIdx, segLen, inf) ==
+	CurrSeg(raw, offset, currInfIdx, currHfIdx, segLen, MetaLen)
+decreases
+func CurrSegEqLemma(raw []byte, offset int, currInfIdx int, currHfIdx int, segLen int) {
+	infOffset := MetaLen + currInfIdx*path.InfoLen
+	inf := reveal path.BytesToIntermediateAbsInfoField(raw[infOffset:infOffset+path.InfoLen], 0, 0, path.InfoLen)
+	unfold acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), R56)
+	unfold acc(sl.AbsSlice_Bytes(raw[infOffset:infOffset+path.InfoLen], 0, path.InfoLen), R56)
+	assert reveal path.BytesToIntermediateAbsInfoField(raw[infOffset:infOffset+path.InfoLen], 0, 0, path.InfoLen) ==
+		reveal path.BytesToIntermediateAbsInfoField(raw, 0, infOffset, len(raw))
+	reveal CurrSeg(raw, offset, currInfIdx, currHfIdx, segLen, MetaLen)
+	reveal CurrSegWithInf(raw[offset:offset+segLen*path.HopLen], currHfIdx, segLen, inf)
+	fold acc(sl.AbsSlice_Bytes(raw, 0, len(raw)), R56)
+	fold acc(sl.AbsSlice_Bytes(raw[infOffset:infOffset+path.InfoLen], 0, path.InfoLen), R56)
+	widenSegment(raw, offset, currHfIdx, inf.AInfo, inf.UInfo, inf.ConsDir, inf.Peer, segLen, offset, offset+segLen*path.HopLen)
+}
+
```

Comment:
let's put all definitions that have to do with `SetInfoField` in a separate file to make sure things are still easy to navigate. Also, these contracts are very hard to read and to understand what is going on. Before merging, please try to abstract the lengthier assertions into new functions. Also, please try to document the purpose of the lemmas informally.